packaging fixes

infinity0 opened the pull request:
I am updating this package in Debian and along the way have made some vendor-agnostic fixes. Please review and merge.

dfries committed:
typo fix, unused variable cleanup, etc

motion.c bad spelling, to "an image"
webhttpd.c consolidate the timeout to the top of the file as I needed
to change it for testing

dfries committed:
set motion capture thread running in the main thread

I identified this while debugging, the thread was created, but hadn't
yet set its state to running before the main thread checked the running
variable and started another thread for the same device.  That
resulted in a crash.  Set running in the main thread, to avoid this
race condition.

dfries committed:
count webcontrol as a thread to avoid a crash

On a SIGHUP restart the main thread waits to restart until all worker
threads are finished executing, except webcontrol wasn't included.
If it was still running (such as reading a web request), it would
have a dangling context pointer after cnt_list was freed leading to a
crash.  This counts that thread in the running threads, as a
termination webcontrol_finish variable to terminate it indepdent of
the device thread, and creates a webcontrol_running to identify when
it is running.

dfries committed:
accept a width specifier to mystrftime

Allow the user to specify field widths to keep the strings from
jumping around as the width changes.  This makes the values easier
to read.

dfries committed:
add power_line_frequency configuration item

The USB webcam I have defaults to the wrong setting causing visible
flickering degrading the image quality.  This allows setting the
value to any of the currently available options or -1 to by default
not modify the value.

dfries committed:
add ffmpeg_duplicate_frames option

This lets the user decide which video problems they would rather see.
In my case a Raspberry Pi 2 with a 640x480 webcam sometimes can keep
up with 10fps so I don't want to set the framerate any lower, but
then sometimes it can't and with duplicate frames it looks like the
video output freezes every second.

rgaufman committed:
Update netcam.c

rgaufman committed:
Update netcam.h

rgaufman committed:
Handle junk data at beginning of JPEG images

bjornfor commented:
+1 from a non-developer trying to package Motion for NixOS :-)

PS: I think a variant of that last commit ("fix build for non-x86 platforms [...]") has been merged to master.

aklomp committed:
bugfix: motion.c: calculate proper allocation size

The proper idiom for calculating the size for memory allocation is:

  ptr = malloc(sizeof(*ptr));

The sizeof() dereferences the pointer's type, and allocates enough
memory to store an instance of that type. motion.c was using this idiom:

  ptr = malloc(sizeof(ptr));

This is incorrect, but thankfully fairly harmless in practice since the
pointer type is usually quite large. Change this to the proper idiom.

aklomp committed:
Bugfix: calculate proper size for memset()

The proper idiom is:

  memset(ptr, val, sizeof(*ptr));

These files were using:

  memset(ptr, val, sizeof(ptr));

dfries committed:
alg_labeling give diffs feedback even when not over the threshold

This is especially useful when picking a threshold to see how
close to the threshold it was.

dfries committed:
redo watchdog termination logic, wait for the thread to cleanup

I apparently have some marginal USB hardware and motion has been
crashing every couple of days in the memcpy at the end of alg.c as
both cnt->imgs.ref and cnt->imgs.image_virgin were NULL pointers.
This was just after the main thread watchdog timer called
pthread_cancel on that thread.  The problem is pthread_cancel can't
possibly kill a thread running on another CPU atomically, although in
this case it's a single core processor and I think pthread_cancel was
releasing it from the ioctl and it would then run to completition.

This modifies the code to not cleanup that content's resources until
that thread is no longer running.  If it runs to completition a normal
restart will happen, if it doesn't the main thread will check once a
second until it no longer is running, cleanup and restart, but it
can't cleanup with that thread running.

dfries committed:
get the thread out of the xioctl loop

As long as errno is EINTR (and that must be the case when the USB
device is still there, but the transfers are failing), the thread
keeps looping running the ioctl when it isn't going to succeed, so
give it access to the finish variable to only loop if the thread isn't
supposed to be going away, and then keep sendig SIGVTALRM to break out
of the ioctl when the thread is supposed to be exiting.  With this fix
the webcam was no longer crashing, which let the webcam without a
problem continue to stream.

dfries committed:
fix dangling pointer cnt->current_image after resize

cnt->current_image because a dangling pointer after image_ring_resize
because it is pointing to cnt->imgs.image_ring which is reallocated in
that routine.  motion_loop will then store cnt->current_image in
old_image which it can then read from.

Reallocations are rare, once in init to size 1, then once to the final
size.  I apparently have a bad USB link and I was seeing a crash
pointing to bad data, after that camera started, then had an error and
crashed in process_image_ring(cnt, IMAGE_BUFFER_FLUSH);
it hadn't yet resized to the normal ring buffer size.  That got me
trying valgrind with a ring buffer size limit of 1 which found this
bug.

infinity0 committed:
packaging fixes from infinity0

Mr-Dave committed:
Additional packaging fixes

Mr-Dave committed:
Add support to ffmpeg.c for libav10

Mr-Dave committed:
Clean up ffmpeg.c and plug memory leak

Mr-Dave committed:
Add copyright file for added features since version 3.2.12

tosiara committed:
Release 3.4.0

tosiara committed:
Added 3.4 to CHANGELOG

Mr-Dave committed:
Merge pull request #37 from tosiara/mrdave-unstable

Added 3.4 to CHANGELOG

Mr-Dave committed:
Add copyright to the 3.4 branch

Mr-Dave committed:
Merge pull request #38 from tosiara/mrdave-motion-3.4

Release 3.4.0

Mr-Dave committed:
Timelapse fixes and bump version

Mr-Dave committed:
Merge pull request #41 from aklomp/mrdave-sizeof

Fix incorrect sizeof() calculations in malloc and memset

sthen committed:
support -b to force running in background

tosiara committed:
Optimization of function mjpegtoyuv420p
Patch proposed by Github user aklomp
Link: https://github.com/Mr-Dave/motion/issues/49

aklomp committed:
mymalloc(): don't cast return value; void* is always promoted

aklomp committed:
mymalloc sets allocated memory to zero; no need for extra memset

aklomp committed:
netcam.c: tfile_context(): simplify function

This function always returned `ret`, so might as well take a shortcut
and return the pointer directly.

aklomp committed:
free(): remove unnecessary NULL pointer checks

Calling free() on a NULL pointer is a completely harmless. Reduce code
complexity by removing the unnecessary checks.

Mr-Dave committed:
Merge pull request #48 from sthen/master

support -b to force running in background

Mr-Dave committed:
Merge pull request #50 from aklomp/mymalloc

Tweaks to uses of 'mymalloc'

Mr-Dave committed:
Update CHANGELOG

Mr-Dave committed:
Fix compile without ffmpeg and older versions

Mr-Dave committed:
Merge unstable with master

Mr-Dave committed:
Updated configure script

Mr-Dave committed:
Allow netcams with modulo 8 dimensions

Mr-Dave committed:
Merge pull request #51 from tosiara/unstable

Optimization of function mjpegtoyuv420p

sthen committed:
forward definition for setup_signals(), needed on OpenBSD

sthen committed:
include rotate.h (for pthread.h) before netcam_rtsp.h

sthen committed:
check for V4L2 header in sys/videoio.h as well as linux/videodev.h

sthen committed:
ioctl(2) "request" is an unsigned long on OpenBSD

sthen committed:
don't detect any *BSD as FreeBSD

sthen committed:
add comment to config file, need to use input 0 on OpenBSD

Mr-Dave committed:
Merge branch 'sthen-openbsd_v4l' into unstable

Mr-Dave committed:
Update changelog

Mr-Dave committed:
Revise version script

Mr-Dave committed:
Bump configure script

Mr-Dave committed:
Lock thread for ffmpeg/libav init

Mr-Dave committed:
bump configure

mterzo committed:
Moved #define events to an enum and added a to string option

tosiara committed:
Fix typo. EMR is not a valid log level. Should be EMG

Mr-Dave committed:
Merge branch 'tosiara-mrdave-fix-typo' into unstable

Mr-Dave committed:
Bump Configure

tosiara committed:
Fix for ftp_send_type - was missing type param
Author: mterzo
https://github.com/mterzo/motion/commit/2dbff19b9dc1a6505e1a279bc210d44259bd789d

tosiara committed:
Typo fix, was build break

tosiara committed:
Typo fix. ERR was mentioned twice, while it should have been INF

Mr-Dave committed:
Fix FTP option

Mr-Dave committed:
Bump configure

Mr-Dave committed:
Merge pull request #61 from tosiara/logging-typo

Typo fix. ERR was mentioned twice, while it should have been INF

jdodgen committed:
Sqlite3 changes - use local sqlite3.c if found. new option to set busy_timeout (wait on locked condition)

jdodgen committed:
V4L2 missing a #include

jdodgen committed:
lastsnap allow path information when doing a singelton lastsnap picture

jdodgen committed:
moved declarations up

jdodgen committed:
replacing spaces with tabs

jdodgen committed:
more spaces to tabs

jdodgen committed:
whoops did one wrong now tab to space conversion :)

jdodgen committed:
Another tab to space change

jdodgen committed:
found another tab that needed spaces

jdodgen committed:
still trying to figure out how may spaces it had

jdodgen committed:
this is a tab in the original spaces to tabs

jdodgen committed:
Whew found it a trailing tab

jdodgen committed:
Tab to space conversion

jdodgen committed:
Remove tabs

jdodgen committed:
Seemed to have lost a few of my changes during my breakapart of patches
Also some formating (tab to spaces) fixes were done

jdodgen committed:
More sqlite changes, now using threaded serial option if sqlite is configured for threads

jdodgen committed:
Fixed problem not finding local sqlite3.h file wne local files are used

jdodgen committed:
Fixed quites in include

jdodgen committed:
reverting tab to spaces changes

jdodgen committed:
another tab reversion

jdodgen committed:
reverting motion.h tab to spaces fix

jdodgen committed:
reverting tab to space

jdodgen committed:
reverting tab to space

jdodgen committed:
Reverting tab to spaces fix

jdodgen committed:
Added ifdef around shared sqlite handle assignment

infinity0 committed:
To include a linux-specific header like linux/types.h, one should test for the presence of linux rather than the absence of BSD.

infinity0 committed:
Fix build for Debian GNU/kFreeBSD which defines __FreeBSD_kernel__ but not BSD

infinity0 commented:
added some commits to build on Debian GNU/kFreeBSD.

as a side note, @bjornfor the variant of 93e214f8 that was commited to master (ad7e243f) is not the appropriate fix. in general if you want to use a feature (there, emms) you should test for its presence (i.e. **MMX**) rather than the lack of opposite features.

jdodgen committed:
Fixed mull pointer causing segfault

dfries committed:
ftp_send_type send type like the function comment says

The function comment says it will send 'I' or 'A', but it was
only sending 'I', even though the function did everything, but
put it in the snprintf (which could be skipped for a fixed string).

Mr-Dave committed:
ffmpeg.c fix GOP size

Mr-Dave committed:
Merge branch 'mterzo-master' of https://github.com/tosiara/motion into tosiara-mterzo-master

Mr-Dave committed:
Merge branch 'unstable' of https://github.com/dfries/motion into dfries-unstable

Mr-Dave committed:
Merge branch 'dfries-unstable' into unstable

Mr-Dave committed:
event.c cleanup

Mr-Dave committed:
Revision for clean build

Mr-Dave committed:
Add the new options to motion-dist.conf

Mr-Dave committed:
Revise install instructions

Mr-Dave committed:
Merge branch 'sqlite-patch' of https://github.com/jdodgen/motion into jdodgen-sqlite-patch

Mr-Dave committed:
Merge branch 'jdodgen-sqlite-patch' into unstable

Mr-Dave committed:
Merge branch 'v4l2-patch' of https://github.com/jdodgen/motion into jdodgen-v4l2-patch

Mr-Dave committed:
Merge branch 'jdodgen-v4l2-patch' into unstable

Mr-Dave committed:
Add revisions to CHANGELOG

Mr-Dave committed:
Logitech MJPEG patch

Mr-Dave committed:
Merge branch 'lastsnap' of https://github.com/jdodgen/motion into jdodgen-lastsnap

Mr-Dave committed:
Merge branch 'jdodgen-lastsnap' into unstable

Mr-Dave committed:
Update CHANGELOG

Mr-Dave committed:
Changelog for hackeron fix for SOI on Netcams

Mr-Dave committed:
Updated gitignore

Mr-Dave committed:
Add additional container options

Mr-Dave committed:
Remove depreciated deinterlace option

Mr-Dave committed:
Ffmpeg container fixes

Mr-Dave committed:
Add preview to webcontrol

Mr-Dave committed:
Remote preview fixes

Mr-Dave committed:
Clean tabs for issue 071831 patch

Mr-Dave committed:
Update manual

Mr-Dave committed:
Fix send raw error introduced with Chrome preview fix

Mr-Dave committed:
Revised motion guide

Mr-Dave committed:
Revise Sqlite3 into code standard

Mr-Dave committed:
Update to motion guide

Mr-Dave committed:
Patch for bad image handling

Mr-Dave committed:
Patch netcam url with proxy

laeos committed:
compile against ffmpeg from github@58fe57d5a05ad8d15c320b2bd163debbc412bdd3

fix minor compilation issues
use pkg-config

laeos committed:
compile against ffmpeg from github@58fe57d5a05ad8d15c320b2bd163debbc412bdd3

fix minor compilation issues
use pkg-config

Mr-Dave committed:
Fix PIX format

Mr-Dave committed:
PIX change for older ffmpeg

Mr-Dave committed:
Slight Packaging Changes

Mr-Dave committed:
Change default in configure for ffmpeg

Mr-Dave committed:
Revised description in configuration file

momo-i committed:
HEVC testing

Mr-Dave committed:
Make/configure changes

momo-i committed:
Merge remote-tracking branch 'upstream/master'

Mr-Dave committed:
HEVC Revision for old ffmpeg

Mr-Dave committed:
Remove ffmpeg coded_frame

Mr-Dave committed:
Add : in netcam_url path spec

Mr-Dave committed:
Swap PKG_CONFIG_PATH and user spec

Mr-Dave committed:
Revised instructions

Mr-Dave committed:
Merge master into 3.4

Mr-Dave committed:
revise version number to 3.4.1

jogu committed:
Produce more helpful error if pkg-config not installed

Currently if users run autoreconf then configure without having
pkg-config installed, they see:

./configure: line 5283: syntax error near unexpected token FFMPEG,'
./configure: line 5283:    PKG_CHECK_MODULES(FFMPEG, libavutil
libavformat libavcodec libswscale, HAVE_FFMPEG=yes)'

This is not hugely helpful and causes support queries and confusion.
It is because the pkg.m4 file that defines PKG_CHECK_MODULES is not
available of pkg-config is not installed.

To work around this, remove PKG_CHECK_MODULES and do roughly the
equivalent calling pkg-config directly. To mirror the current
behaviour, we also add a test for pkg-config being present and abort
if it is not. (I am not 100% sure this is the desired behaviour, but
regardless this commit is still an improvement on what we currently
have.)

jogu committed:
Fix stack corruption in event_ffmpeg_timelapse

The current definitions (char codec_swf[3] = "swf" etc) are declared
one character too short resulting in stack corruption or crashes in
some cases.

To avoid the possibility of an error at all, we change ffmpeg_open to
take a const char * as it doesn't need to alter the string.

Closes https://github.com/Mr-Dave/motion/issues/109

ccrisan committed:
h264 codec: set crf option based on configured vbr

ccrisan committed:
h264 codec: fixed crf from vbr: the vbr range is actually 1 to 32767

jogu committed:
Fix bad html in server home page

</a/n> was presumably meant to be a </a> followed by a new line.

ccrisan committed:
ffmpeg/h264: actually pass the crf value to the options

jogu committed:
#104 Fix compile failing on raspberry pi 3

When we test if the compiler options work, we didn't test for the
possibility of the compiler crashing which unfortunately it does on
the raspberry pi 3 - crashing in this case means returning exit code
134 (i.e. SIGABRT).

Change the test so that instead of testing for one particular failure,
we invert and test for success - treating everything else as failure.

jogu committed:
Fix #104 Use optimal compilation options on Raspberry Pi 3

gcc on the raspberry pi 3 crashes when called from our configure
script, and even in simpler cases:

$ gcc -mcpu=native -march=native
*** Error in `gcc': double free or corruption (top): 0x0190dbc0 ***
Aborted

So we need to add explicit optimisation options if we want to build an
optimal binary on the Pi 3.

As per:

https://www.element14.com/community/community/raspberry-pi/blog/2016/02/29/the-most-comprehensive-raspberry-pi-comparison-benchmark-ever

the raspberry pi 3 is an armv8, but cpuinfo incorrectly reports it as
an armv7. So we can’t detect the CPU using /proc/cpuinfo.

Instead we use /proc/device-tree/model, which explicit has the
Raspberry PI type in it, e.g. “Raspberry Pi 3 Model B Rev 1.2”.

https://www.raspberrypi.org/forums/viewtopic.php?t=144115&p=952479
lists hopefully correct optimal gcc flags, namely:

Some basic options that should be the defaults anyway:-
-mfloat-abi=hard -mlittle-endian -munaligned-access
(confirmed; these are the defaults)

Pi1 -mcpu=arm1176jzf-s  -mfpu=vfp
Pi2 -mcpu=cortex-a7  -mfpu=neon-vfpv4
Pi3 -mcpu=cortex-a53  -mfpu=neon-fp-armv8

Mr-Dave committed:
Revise version number

Mr-Dave committed:
Merge branch 'stack-corruption' of https://github.com/jogu/motion into jogu-stack-corruption

Mr-Dave committed:
Merge branch 'jogu-stack-corruption'

Mr-Dave committed:
Change log update

Mr-DaveDev committed:
Updated configure.ac for pkg-config

Mr-Dave committed:
Merge pull request #138 from Mr-DaveDev/pkg-config

Revised configure messages for pkg-config

Mr-DaveDev committed:
Merge branch 'master' of https://github.com/ccrisan/motion into ffmpeg-fixes

Mr-DaveDev committed:
FPS Fix.  Issue 132

Mr-DaveDev committed:
ffmpeg 3.0 fixes

Mr-DaveDev committed:
Eliminate developer-flags warnings

Mr-DaveDev committed:
Fix truncate images. Issue 131

Mr-Dave committed:
Merge pull request #140 from jogu/rasppi3-fix

Fix for gcc crashing during configure / unable to build on raspberry pi 3

Mr-DaveDev committed:
Scale movie quality/crf values based upon ffmpeg_variable_bitrate values from 0 to 100

Mr-DaveDev committed:
Report versions used and revise log levels

Mr-DaveDev committed:
Merge remote-tracking branch 'upstream/master' into ffmpeg-fixes

Mr-DaveDev committed:
ffmpeg log messages and init

Mr-Dave committed:
Merge pull request #141 from jogu/html-fix

Fix bad html in server home page

Mr-DaveDev committed:
Timelapse Container Fix

Mr-DaveDev committed:
Timelapse revision of swf to mpg

Mr-DaveDev committed:
Documentation Update

Mr-DaveDev committed:
PTS fixes

tosiara committed:
Updated extpipe examples with ultrafast H264 (ffmpeg, mencoder and x264)
Also updated previous example to use %w and %h variables

Mr-DaveDev committed:
PTS refinements, add test and remove ogg options

Mr-DaveDev committed:
Better implementation of the test codec

Mr-DaveDev committed:
Change old containers to defaults

Mr-Dave committed:
Merge pull request #150 from Mr-DaveDev/ffmpeg-fixes

ffmpeg fixes

Mr-DaveDev committed:
Timelapse fps and regression fixes

Mr-DaveDev committed:
Revert to standard os file buffering

Mr-Dave committed:
Merge pull request #153 from Mr-DaveDev/image-corruption

Revert to standard os file buffering

jogu committed:
Initial support for MMAL camera on raspberry pi

Originally by dozencrows on github; mostly this commit:

https://github.com/dozencrows/motion/commit/6d1ed5bd011d83b6358daf86a20c60cb50d0418c

Changes to merge back into motion upstream made by Joseph Heenan (with
help from lowflyerUK on github, see https://github.com/dozencrows/motion/pull/1 ):

- changed back to using autoconf etc as per the main project
- submitted the one change we need to the raspicam files back upstream
  ( https://github.com/raspberrypi/userland/pull/332 )
- imported latest versions of files in raspicam, now exactly match upstream
- added README.txt to raspicam directory with brief explanation
- replaced some tabs with spaces
- removed dependency on checkout of raspberrypi userland repo; now we
  use the headers provided by libraspberrypi-dev
- merged in a couple of the trivial later bugfix commits
- merged in mmalcam_control_params commit to allow config of camera options
- fixed many merge conflicts rebased on top of latest upstream motion

Mr-DaveDev committed:
Motion guide update to reference release page and fix some wording

Mr-Dave committed:
Merge pull request #152 from Mr-DaveDev/ffmpeg-fixes

Timelapse fps and regression fixes

mterzo committed:
Adding travis-ci to test build code

Mr-Dave committed:
Merge pull request #151 from Mr-DaveDev/doc-updates

Documentation Update

jogu committed:
Fix build for libavutil prior to 51.42.0

jogu committed:
Make some internal functions "static"

(and remove the places they were added to .h files)

I don't believe these functions were intended to be exposed outside
the .c file they're defined in, so make them internal functions.

jogu committed:
Add camera_name setting to allow each camera to be given a name

Based on a commit originally by Chris Roberts <86montecarlo@gmail.com>

camera_name allows naming of threads, is displayed in the motion control
webpage, and a new specifier (%$) added enabling it to be used for
filenames and such.

The changes I have made since Chris's commit:

1. Change format specifier for threadname from %E to %$

This avoids clashing with any known strftime format specifiers.
(According to the Debian stable strftime man page - it seems we've
already used all spare letters to mean other things.)

2. Behaviour if camera_name is not set is now unchanged

(Previous it would display "thread 1 -- NULL" if the thread name was
unset, and I don't believe how printf deals with NULL values for %s
specifiers is guaranteed.)

3. Thread name is reflected into the system thread name

Uses pthread_setname_np. There's a autoconf test for it, which looks
for the version that takes the thread id and thread name. I believe
this should work on all recent linux, and maybe other platforms too.

(It won't work on macOS; autoconf will disable it - but I deliberately
always use the pthread_setname_np(pthread_self(), name) form so that
it could easily be adapted for macOS in the future.)

4. The parameter has been renamed from thread_name to camera_name, as
that seems to better reflect how the user will see it.

jogu committed:
Merge pull request #156 from jogu/travis

Fix for older libavutil + add travis configuration

Mr-Dave committed:
Merge pull request #157 from jogu/compile-warnings-fix

Make some internal functions "static"

Mr-Dave committed:
Merge pull request #148 from jogu/mmal-initial

Initial support for MMAL camera on raspberry pi

jogu committed:
Stop "make clean" deleting documentation jpgs

make clean currently removes all jpgs, which means it deletes the jpgs
used in the documentation (outputnormal1.jpg etc). This seems like it
wasn't intended.

I can't figure out why it deletes jpgs at all, it seems to date back
to the initial import of version 3.2.5 into svn in 2005 - seems safe
to remove it.

jogu committed:
Fix for whitespace lines in config file

The while loop was missing a condition to stop at the start of the line,
so we'd carry on beyond the start of the buffer.

Closes #88

jogu committed:
Possible fix for failure to connect to mysql on startup

As per http://dev.mysql.com/doc/refman/5.7/en/mysql-library-init.html
we should init the mysql library before starting any threads, as
otherwise mysql_init is not thread safe.

Closes #103

jogu committed:
Fix cases where gethostname() returns a name that doesn't work for client

We currently rely on gethostname() to generate the links from the web
control homepage to the camera streams - unfortunately if DNS isn't
set up well the hostname won't be one the web browser can resolve
(often the case on raspberry pi's, where the hostname is set to
"raspberrypi" by default and users access them via IP address).

To cater for these situations we look at the "Host:" header in the
HTTP headers that came from the browser (which indicate which host the
browser has been asked to connect to) and use that to generate the
hostname to use in the links.

If there's no Host: header (unlikely these days) we default back to
the old behaviour.

Mr-Dave committed:
Merge pull request #160 from jogu/make-clean-fix

Stop "make clean" deleting documentation jpgs

Mr-Dave committed:
Merge pull request #161 from jogu/whitespace-only-conf-lines

Fix for whitespace only lines in config file

Mr-Dave committed:
Merge pull request #163 from jogu/workaround-for-gethostname

Fix cases where gethostname() returns a name that doesn't work for client

Mr-Dave committed:
Merge pull request #158 from jogu/camera_name

Add camera_name setting to allow each camera to be given a name

Mr-Dave committed:
Merge pull request #162 from jogu/mysql-thread-safe-init

Possible fix for failure to connect to mysql on startup

Mr-DaveDev committed:
Revise the condition for the MYPixelFormat to use the AVFORMAT version number

Mr-DaveDev committed:
Suppress excessive and unfixable avcodec messages

Mr-DaveDev committed:
Remove unused variables of ostatus and loop

jogu committed:
Merge pull request #166 from Mr-DaveDev/old-ffmpeg

MYPixelFormat Revision

jogu committed:
Merge pull request #168 from Mr-DaveDev/loopback-warnings

Remove unused variables in vloopback_motion

jogu committed:
Merge pull request #167 from Mr-DaveDev/netcam-messages

Suppress some messages from avcodec

mterzo committed:
Adding ubuntu 14.04 to build along with 12.04 in travis-ci

mterzo committed:
Renamed thread to camera to more clearly define what
thread is.  Keeping backwards compatability with
thread so that old configs can still be used.

ffontaine committed:
Fix jpeg-turbo detection

Do not only check for static libjpeg.a but also for dynamic libjpeg.so

Signed-off-by: Fabrice Fontaine <fabrice.fontaine@orange.com>

jogu committed:
Merge pull request #173 from ffontaine/master

Fix jpeg-turbo detection

jogu committed:
Merge pull request #170 from mterzo/tavis-ci-trusty

Adding ubuntu 14.04 to build along with 12.04 in travis-ci

ffontaine committed:
Fix detection of sqlite3 when cross-compiling

Replace usage of AC_CHECK_FILE by test -f to check the existence of
sqlite3.c as AC_CHECK_FILE does not work when cross-compiling

Signed-off-by: Fabrice Fontaine <fabrice.fontaine@orange.com>

ccrisan committed:
automatically resize mask image if size differs from that of the captured image

jogu committed:
Fix for first captured picture after start up is zero length

If the first fopen() failed, then myfopen() always returned NULL even if
the second fopen() succeeded. Now it correctly returns the result of the
second fopen().

Authored by https://github.com/pcwalden

Closes #155

jogu committed:
Merge pull request #174 from ffontaine/master

Fix detection of sqlite3 when cross-compiling

ffontaine committed:
Change --without-sdl to --with-sdl=[DIR]

With this modification, the user will be able to specify the path to
sdl-config which can be outside his path (for example on embedded
buildsystem such as buildroot)

Signed-off-by: Fabrice Fontaine <fabrice.fontaine@orange.com>

ffontaine committed:
Fix jpeg-turbo detection

Commit number 4998d465f7837a01c02d88e6d213e5f881e62122 added the detection of jpeg-turbo dynamic library through -o but -f argument was missing so test was always true

Signed-off-by: Fabrice Fontaine <fabrice.fontaine@orange.com>

jogu committed:
Merge pull request #180 from ffontaine/jpeg-turbo-fixes

Fix jpeg-turbo detection

ffontaine committed:
Fix LIBS and CFLAGS when jpeg-turbo is enabled

When jpeg-turbo was enabled, LIBS was uncorrectly set: -lpthread, -lm,
... was lost. Same issue with CFLAGS.

Fixes #17.

Signed-off-by: Fabrice Fontaine <fabrice.fontaine@orange.com>

tosiara committed:
Updated motion_guide.htm with extpipe description and examples

Mr-DaveDev committed:
Merge branch 'camera_id' of https://github.com/mterzo/motion into mterzo-camera_id

jogu committed:
Merge pull request #181 from ffontaine/jpeg-turbo-fixes

Fix LIBS and CFLAGS when jpeg-turbo is enabled

slokhorst committed:
Fix hostname determination for IPv6 addresses in web server.

slokhorst committed:
Fix string length in memchr call.

slokhorst committed:
Cleanup HTML.

slokhorst committed:
Fix another hostname issue, where the hostname would contain the next header line, when the
Hostname header was empty or malformed.

pcwalden committed:
Adds --without-mmal option to configure.
As documented in issue 169, the recentlay added MMAL camera support for the
Raspberry Pi does not build on the Wheezy version of Raspbian.
This option disables the autoconf generation of MMAL dependencies.

Mr-DaveDev committed:
Additional changes for move to camera instead of thread

Mr-DaveDev committed:
Log Level for Mask and Memory Leak

Revise the log level to warning when the user has specified a pgm mask file that is
different than the size of the image from the camera.

Free the memory associated with mask image to close memory leak.

Mr-DaveDev committed:
Merge branch 'ccrisan-master' into mask_resize

Mr-Dave committed:
Merge pull request #176 from ffontaine/master

Change --without-sdl to --with-sdl=[DIR]

Mr-Dave committed:
Merge pull request #187 from Mr-DaveDev/camera_id

Camera ID

Mr-Dave committed:
Merge pull request #188 from Mr-DaveDev/mask_resize

Mask resize

Mr-Dave committed:
Merge pull request #179 from tosiara/extpipe-update-examples

Updated extpipe examples with ultrafast H264 (ffmpeg, mencoder and x264)

jogu committed:
Merge pull request #182 from slokhorst/ipv6-hostname

Fix hostname determination for IPv6 addresses in web server.

jogu committed:
Merge pull request #186 from pcwalden/master

Adds --without-mmal option to configure.

jogu committed:
Merge pull request #184 from slokhorst/html-cleanup

Cleanup HTML.

slokhorst committed:
Add systemd service file
Remove Fedora initscript (use the systemd service file instead)

mterzo committed:
Camera configuration directory

Mr-Dave committed:
Merge pull request #189 from mterzo/camera_conf_dir

Camera configuration directory

Mr-Dave committed:
Merge pull request #185 from slokhorst/systemd

Add systemd service file

jogu committed:
Fix for a couple of warnings in developer builds

jogu committed:
Prevent blank line after ffmpeg log entries

ffmpeg seems to put a \n on the end of messages, and MOTION_LOG also
adds one, so remove the ffmpeg one.

jogu committed:
Update .gitignore for thread -> camera rename of config files

slokhorst committed:
Remove old pwc-ioctl.h files, which are not used anywhere in the program

slokhorst committed:
Remove README.axis_2100 file which only has trivial instructions

Mr-Dave committed:
Merge pull request #195 from jogu/misc-tweaks

Small fixes for recent changes

Mr-Dave committed:
Merge pull request #198 from slokhorst/rm-old-files

Remove old files

infinity0 committed:
Fix line endings for man pages

man pages are typically for UNIX systems where the line endings are LF not CRLF.
The file currently uses a mix of both, urgh.

infinity0 committed:
Fix paths in documentation

Motion already installs motion-dist.conf to @sysconfdir@/motion/, but some
documentation was not updated to reflect this yet.

infinity0 committed:
Fix spelling/grammar mistakes

infinity0 committed:
Fix some under-escaping in motion.1 manpage

Certain symbols like - and \ need to be escaped in the man format
